### ISSUE TITLE
isinstance with `Protocol` fails in doctest

### ISSUE THREAD

**Issue by nstarman** [simplistix/sybil#51]
In some docs I tried showing `isinstance(x, Protocol)` but this fails, complaining that `"__name__"` isn't in in the function's global keys. I believe this is a consequence of how Sybil sets up the doc tests, though I haven't been able to diagnose the specific issue. Other than this, Sybil has been great, thanks for the package!

Traceback
```
Example at /Users/.../docs/introduction.rst, line 171, column 1 did not evaluate as expected:
Exception raised:
    Traceback (most recent call last):
      File "/Users/.../lib/python3.10/doctest.py", line 1350, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest /Users/.../docs/introduction.rst[0]>", line 1, in <module>
        isinstance(v, NamedValue)
      File "/Users/.../lib/python3.10/typing.py", line 1496, in __instancecheck__
        not _allow_reckless_class_checks(depth=2)
      File "/Users/.../lib/python3.10/typing.py", line 1473, in _allow_reckless_class_checks
        return sys._getframe(depth).f_globals['__name__'] in ['abc', 'functools']
    KeyError: '__name__'
```

MWE
```
.. code-block:: python

    from typing import Protocol


    class NamedValue(Protocol):
        """Interface for an object that has a name and a value."""

        value: float
        name: str

some text

.. code-block:: python

    class NamedValueClass1:
        def __init__(self, name: str, value: float):
            self.name = name
            self.value = value


    v = NamedValueClass1("foo", 1.0)

::

    >>> isinstance(v, NamedValue)
    True
```

**Comment by cjw296** [comment]
This is fixed in https://github.com/simplistix/sybil/commit/2b7711c10da12a4551c7599e495f6d4a9637d866.

**Comment by nstarman** [comment]
Thanks!!

**Comment by cjw296** [comment]
Released in 5.0.2
